### DISCUSSION TITLE
docs/reference/cli

### DISCUSSION THREAD

**Discussion by giscus** [zenstackhq/zenstack-docs#364] [Comments]
# docs/reference/cli

CLI references

https://zenstack.dev/docs/reference/cli

**Comment by tmax22** [comment]
how to specify plugin names? for example, lets say i want to run only 'zod' plugin. its still running all plugins:
```
❯ zenstack  generate --with-plugins zod                                                                                                                                                                                                                                                                        ─╯
⌛️ ZenStack CLI v2.5.1, running plugins
✔ Generating Prisma schema
✔ Generating PrismaClient enhancer
✔ Generating Zod schemas

```

**Reply by ymc9** [reply]
Good question. The `--with-plugins` and `--without-plugins` only governs explicitly declared plugins in ZModel. If you want to disable a built-in plugin like zod, there are two ways:

1. Explicitly declare it (which takes precedence over the built-in one) then exclude it.
2. Use the "--no-default-plugins" to disable auto-inclusion of all built-in plugins. But you'll need to manually declare other built-in ones ("@core/prisma", "@core/enhancer") to make ZenStack function properly.

Just curious, why do you want to explicitly exclude the zod plugin?

**Reply by tmax22** [reply]
i have zod in schema like so:
```zmodel
...
plugin zod {
    provider = '@core/zod'
}
...
```
the `zenstack  generate --with-plugins zod` command seems to run all (the docs states for `--with-plugins` option that `only run specific plugins`) plugins including schema and enhancer generation.

about your question
>why do you want to explicitly exclude the zod plugin

it was just for example. a real use case is running only Prisma schema generation for quickly testing local. currently in such cases i usually edit schema.prisma file manually quickly and when i finish i copy to zmodel and generate all. How would I run schema generation only?

**Reply by ymc9** [reply]
I think you can declare a "@core/prisma" plugin and then use "--with-plugins" to run it only.

**Comment by PodkopovP** [comment]
To enable support with the new prisma-client provider, zenstack should import from the same location as the "output" declared in the generator - as it's currently trying to import from "@prisma/client", which isn't (and probably wont be) where people decide to output their prisma client

Can't use zenstack at the moment :/

**Reply by ymc9** [reply]
I'm working on a fix in the next release. Please use the old prisma generator for now.
